backend/src/interface/persistent_data/mod.rs

```diff
@@ -1 +1,141 @@
 //! The interfaces specified here allow access to data stored in a persistent datastore like a database.
+use crate::interface::persistent_data::model::{
+    ApiKey, Canteen, DataError, Image, ImageInfo, Line, Meal, Side,
+};
+
+use crate::util::{Additive, Allergen, Date, MealType, Price, ReportReason};
+use async_trait::async_trait;
+use chrono::{DateTime, Local};
+use uuid::Uuid;
+
+mod model;
+
+pub type Result<T> = std::result::Result<T, DataError>;
+
+#[async_trait]
+/// An interface for checking relations and inserting data structures. The MealplanManagement component uses this interface for database access.
+pub trait MealplanManagementDataAccess {
+    /// Determines all canteens with a similar name.
+    async fn get_similar_canteens(similar_name: String) -> Result<Vec<Canteen>>;
+    /// Determines all lines with a similar name.
+    async fn get_similar_lines(similar_name: String) -> Result<Vec<Line>>;
+    /// Determines all meals with a similar name.
+    async fn get_similar_meals(similar_name: String) -> Result<Vec<Meal>>;
+    /// Determines all sides with a similar name.
+    async fn get_similar_sides(similar_name: String) -> Result<Vec<Side>>;
+
+    /// Updates an existing canteen entity in the database. Returns the entity.
+    async fn update_canteen(uuid: Uuid, name: String) -> Result<Canteen>;
+    /// Updates an existing line entity in the database. Returns the entity.
+    async fn update_line(uuid: Uuid, name: String) -> Result<Line>;
+    /// Updates an existing meal entity in the database. Returns the entity.
+    async fn update_meal(
+        uuid: Uuid,
+        line_id: Uuid,
+        date: Date,
+        name: String,
+        price: Price,
+    ) -> Result<Meal>;
+    /// Updates an existing side entity in the database. Returns the entity.
+    async fn update_side(
+        uuid: Uuid,
+        line_id: Uuid,
+        date: Date,
+        name: String,
+        price: Price,
+    ) -> Result<Side>;
+
+    /// Adds a new canteen entity to the database. Returns the new entity.
+    async fn insert_canteen(name: String) -> Result<Canteen>;
+    /// Adds a new line entity to the database. Returns the new entity.
+    async fn insert_line(name: String) -> Result<Line>;
+    /// Adds a new meal entity to the database. Returns the new entity.
+    async fn insert_meal(
+        name: String,
+        c_type: MealType,
+        price: Price,
+        next_served: Date,
+        allergens: Vec<Allergen>,
+        additives: Vec<Additive>,
+    ) -> Result<Meal>;
+    /// Adds a new side entity to the database. Returns the new entity.
+    async fn insert_side(
+        name: String,
+        c_type: MealType,
+        price: Price,
+        next_served: Date,
+        allergens: Vec<Allergen>,
+        additives: Vec<Additive>,
+    ) -> Result<Side>;
+}
```

Comment:
I would suggest putting all these Traits into seperate files as they re quite long.

Reply:
I thought about that, but if we do it, then we do it everywhere. We don't want inconsistent folder/file styles.